pipeline/src/since/stream.rs

```diff
@@ -0,0 +1,213 @@
+        self
+    }
+
+    fn schema(&self) -> SchemaRef {
+        self.source.schema().clone()
+    }
+
+    fn table_type(&self) -> TableType {
+        TableType::View
+    }
+
+    async fn scan(
+        &self,
+        _state: &dyn Session,
+        projection: Option<&Vec<usize>>,
+        filters: &[Expr],
+        limit: Option<usize>,
+    ) -> Result<Arc<dyn ExecutionPlan>, DataFusionError> {
+        let schema = projection
+            .map(|projection| self.schema().project(projection))
+            .transpose()?
+            .map(Arc::new)
+            .unwrap_or_else(|| self.schema());
+        Ok(Arc::new(StreamExec {
+            source: self.source.clone(),
+            projection: projection.cloned(),
+            offset: filters
+                .iter()
+                .filter_map(Self::highwater_mark_from_expr)
+                .next()
+                .map(|hm| hm as i64)
+                .unwrap_or(0),
+            limit,
+            properties: PlanProperties::new(
+                EquivalenceProperties::new(schema),
+                Partitioning::UnknownPartitioning(1),
+                ExecutionMode::Unbounded,
+            ),
+        }))
+    }
+    fn supports_filters_pushdown(
+        &self,
+        filters: &[&Expr],
+    ) -> datafusion::common::Result<Vec<TableProviderFilterPushDown>> {
+        Ok(filters
+            .iter()
+            .map(|expr| Self::highwater_mark_from_expr(expr))
+            .map(|highwater_mark| {
+                if highwater_mark.is_some() {
+                    TableProviderFilterPushDown::Exact
+                } else {
+                    TableProviderFilterPushDown::Unsupported
+                }
+            })
+            .collect())
+    }
+}
+
+#[derive(Debug)]
+pub struct StreamExec<S> {
+    source: S,
+    projection: Option<Vec<usize>>,
+    offset: i64,
+    limit: Option<usize>,
+    properties: PlanProperties,
+}
+
+impl<S: StreamTableSource> ExecutionPlan for StreamExec<S> {
+    fn name(&self) -> &str {
+        "StreamExec"
+    }
+
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn properties(&self) -> &PlanProperties {
+        &self.properties
+    }
+
+    fn children(&self) -> Vec<&Arc<dyn ExecutionPlan>> {
+        vec![]
+    }
+
+    fn with_new_children(
+        self: Arc<Self>,
+        _children: Vec<Arc<dyn ExecutionPlan>>,
+    ) -> datafusion::error::Result<Arc<dyn ExecutionPlan>> {
+        Ok(self)
+    }
+
+    fn execute(
+        &self,
+        _partition: usize,
+        _context: Arc<datafusion::execution::TaskContext>,
+    ) -> datafusion::error::Result<datafusion::execution::SendableRecordBatchStream> {
+        let source = self.source.clone();
+        let projection = self.projection.clone();
+        let offset = self.offset as u64;
+        let limit = self.limit;
+        let stream = try_stream! {
+            let mut stream = source.subscribe_since(projection, Some(offset), limit).await?;
```

Comment:
Here we are inside an `execute` method that gets called during a datafusion query. The implementation calls `subscribe_since` on its source (which is an actor handle).